src/qt/guiutil.cpp

```diff
@@ -858,9 +858,9 @@ void restoreWindowGeometry(const QString& strSetting, const QSize& defaultSize,
     parent->resize(size);
     parent->move(pos);
 
-    if ((!pos.x() && !pos.y()) || (QApplication::desktop()->screenNumber(parent) == -1))
-    {
-        QRect screen = QApplication::desktop()->screenGeometry();
+    int screen_number = QApplication::desktop()->screenNumber(parent);
+    QRect screen = QApplication::desktop()->availableGeometry(screen_number);
+    if ((!pos.x() && !pos.y()) || screen_number == -1 || !screen.contains(parent->frameGeometry())) {
```

Comment:
Does this allow a window that is halfway moved out of the screen (I think we should allow this)?

Reply:
No I don't think it does, but I thought about that and the expected behaviour of most software is that it launches within the screen even if you moved it partially out last time?